R/unidimensionalReliabilityFrequentist.R

```diff
@@ -1369,4 +1678,373 @@ applyomegaPFA <- function(m, callback = function(){}, loadings = FALSE){
+          boot_data <- as.matrix(data[sample.int(n, size = n, replace = TRUE), ])
+          if (pairwise) {
+            fitTmp <- Bayesrel:::fitmodelMis(mod, boot_data)
+          } else {
+            fitTmp <-  Bayesrel:::fitmodel(mod, boot_data)
+          }
+          callback()
+          if (!is.null(fitTmp)) {
+            if (standardized) {
+              params <- lavaan::standardizedsolution(fitTmp, level = interval)
+              om_obj[i] <- params$est.std[params$lhs=="omega"]
+            } else {
+              params <- lavaan::parameterestimates(fitTmp, level = interval)
+              om_obj[i] <- params$est[params$lhs=="omega"]
+            }
+
+          } else {
+            om_obj[i] <- NA
+          }
+        }
+
+        if (sum(!is.na(om_obj)) > 1) {
+          om_low <- quantile(om_obj, prob = (1 - interval) / 2, na.rm = TRUE)
+          om_up <- quantile(om_obj, prob = interval + (1 - interval) / 2, na.rm = TRUE)
+        } else {
+          om_low <- NA
+          om_up <- NA
+          om_obj <- NA
+        }
+      }
+    }
+
+    fit_tmp <- lavaan::fitMeasures(fit)
+    indic <- c(fit_tmp["chisq"], fit_tmp["df"], fit_tmp["pvalue"],
+               fit_tmp["rmsea"], fit_tmp["rmsea.ci.lower"], fit_tmp["rmsea.ci.upper"],
+               fit_tmp["srmr"])
+  }
+  return(list(omega = omega, omega_lower = om_low, omega_upper = om_up, indices = indic, fit.object = fit,
+              omega_boot = om_obj))
+}
+
 
+.applyOmegaCov <- function(
+    cc,
+    missing,
+    callback = function(){})
+  {
+
+  p <- ncol(cc)
+  file <- Bayesrel:::lavOneFile(cc)
+  colnames(cc) <- file$names
+
+  lam_names <- paste("l", 1:p, sep = "")
+  err_names <- paste("e", 1:p, sep = "")
+  model <- paste0("f1 =~ ")
+  loadings <- paste(paste(lam_names, "*", file$names, sep = ""),
+                    collapse = " + ")
+  errors <- paste(paste(file$names, " ~~ ", err_names, "*",
+                        file$names, sep = ""), collapse = "\n")
+  sum_loads <- paste("loading :=", paste(lam_names, collapse = " + "),
+                     "\n")
+  sum_errs <- paste("error :=", paste(err_names, collapse = " + "),
+                    "\n")
+  omega <- "omega := (loading^2) / ((loading^2) + error) \n"
+  mod <- paste(model, loadings, "\n", errors,
+               "\n", sum_loads, sum_errs, omega)
+
+  fit <-  .fitModelCov(mod, sample.cov = cc, sample.nobs = 500, missing)
+
+  callback()
+
+  if (is.null(fit)) {
+    return(NA)
+  } else {
+    params <- lavaan::parameterestimates(fit)
+    omega <- params$est[params$lhs == "omega"]
+  }
+  return(omega)
+}
+
+
+.fitModelCov <- function (mod, sample.cov, sample.nobs = 500, missing = "listwise") {
+  out <- tryCatch({
+    lavaan::cfa(mod, sample.cov = sample.cov, sample.nobs = sample.nobs, std.lv = TRUE, missing = missing)
+  }, error = function(cond) {
+    return(NULL)
+  }, warning = function(cond) {
+    return(NULL)
+  }, finally = {
+  })
+  return(out)
+}
+
+#### Functions from Andries ####
```

Comment:
is there a reference for this? e.g., a paper?

Reply:
see help file